services/collaboration/pkg/connector/fileconnector.go

```diff
@@ -1240,12 +1244,43 @@ func (f *FileConnector) CheckFileInfo(ctx context.Context) (*ConnectorResponse,
 		}
 	}
 
+	// if the file content is empty and a template reference is set, add the template source URL
+	if wopiContext.TemplateReference != nil && statRes.GetInfo().GetSize() == 0 {
+		if tu, err := f.createDownloadURL(wopiContext, collaborationURL); err == nil {
+			infoMap[fileinfo.KeyTemplateSource] = tu
+		}
+	}
+
 	info.SetProperties(infoMap)
 
 	logger.Debug().Interface("FileInfo", info).Msg("CheckFileInfo: success")
 	return NewResponseSuccessBody(info), nil
 }
 
+// createDownloadURL will create a download URL for the template file.
+// It uses a new wopi context with the template reference set as the file reference
+// and a reva access token to download the file.
+func (f *FileConnector) createDownloadURL(wopiContext middleware.WopiContext, collaborationURL *url.URL) (string, error) {
+	templateContext := wopiContext
+	templateContext.FileReference = wopiContext.TemplateReference
+	templateContext.TemplateReference = nil
+
+	token, _, err := middleware.GenerateWopiToken(templateContext, f.cfg)
+	if err != nil {
+		return "", err
+	}
+	downloadURL := *collaborationURL
+	downloadURL.Path = path.Join(
+		collaborationURL.Path,
+		"wopi/templates/",
+		helpers.HashResourceId(templateContext.FileReference.GetResourceId()),
+	)
+	q := downloadURL.Query()
+	q.Add("access_token", token)
```

Comment:
the access token is only passed in the backend. not happy that it will appear in logs as it is part of the url, but we can change this in a subsequent pr when we store the wopi session in nats and use an opaque string in the url.

Reply:
unfortunately many WOPI apps are using the access_token as an URL param, Specially in GET requests.